### PR TITLE
Enhance `plot_pdp` and fix `plot_scatter_submodels`

### PR DESCRIPTION
**Pull request by AlexAndorra**
- Add env YML files, especially for dev
- Add `softmax_link` option to `plot_pdp` for the Categorical likelihood case
- Fix `plot_scatter_submodels` to be able to handle the categorical likelihood case

### CODE REVIEW

File: pymc_bart/utils.py [hunk 1]
```diff
@@ -330,6 +331,9 @@ def plot_pdp(
         List of the indices of the covariate treated as discrete.
     func : Optional[Callable], by default None.
         Arbitrary function to apply to the predictions. Defaults to the identity function.
+    softmax_link: Optional[bool] = False,
```
Comment (aloctavodia) [inline pymc_bart/utils.py:334]: We already have a `func` argument. We should use that argument to pass arbitrary functions, including the softmax.

**Review by aloctavodia — CHANGES_REQUESTED** [review]

File: pymc_bart/utils.py (hunk 1 above)
Comment (AlexAndorra) [reply pymc_bart/utils.py]: Addressed @aloctavodia

File: pymc_bart/utils.py [hunk 2]
```diff
@@ -396,11 +396,18 @@ def identity(x):
         p_d = _sample_posterior(
             all_trees, X=fake_X, rng=rng, size=samples, excluded=excluded, shape=shape
         )
+        # need to apply func to full array and to last dimension if it's softmax
+        if func.__name__ == "softmax":
```
Comment (aloctavodia) [inline pymc_bart/utils.py:400]: either we should ask the user to handle these special cases themselves with something like `lambda x: np.apply_along_axis(link_func, axis=1, arr=x)` or provide a general solution, not a specific solution for the softmax

Comment (AlexAndorra) [reply pymc_bart/utils.py:400]: Mmmh yeah, agreed. I don't think there are that many cases, but you're right we can't anticipate everything users are gonna want to do, so it may be better to ask the user to handle these special cases. 
The issue here is that the function has to be applied upstream compared to a simple `logit` for instance. I gotta see if and how we can handle that

Comment (aloctavodia) [reply pymc_bart/utils.py:400]: The easy way is to leave it unchanged and document it. The other could be to have a `func` argument and a `func_kwargs` argument

Comment (AlexAndorra) [reply pymc_bart/utils.py]: Actually, since it looks like `func` was developed with element-wise functions in mind, we can just apply upstream, at the same place as where we should apply a multidimensional function like softmax. In other words:
```python
# Option 1: Apply function to slice
result1 = func(p_d[:, :, s_i])

# Option 2: Apply function to whole array then slice
result2 = func(p_d)[:, :, s_i]
```
Both results are the same, because element-wise functions apply the same transformation to each element without considering the relationships between elements, so I think that works!

I just pushed that change, so LMK what you think, and if I missed something!

Comment (aloctavodia) [reply pymc_bart/utils.py]: This is right, the same should apply to `plot_ice`, can you change that too?

**Comment by AlexAndorra** [comment]
Thanks for the ping amigo! I didn't forget, I'm just swamped right now, but
will push the change ASAP

**Comment by aloctavodia** [comment]
I just added the change to `plot_ice` myself. You already helped a lot with the other changes. Thanks @AlexAndorra.
